integreat_cms/deepl_api/utils.py

```diff
@@ -0,0 +1,87 @@
+import deepl
+import os
+
+from django.contrib import messages
+
+
+class DeepLApi:
+    """
+    DeepL API to auto translate selected posts.
+    """
+
+    def check_availability(self, request, language_slug):
+        """
+        this function checks, if the selected language is supported by DeepL
+
+        :param request: request that was sent to the server
+        :type request: ~django.http.HttpRequest
+
+        :param language_slug: current language slug
+        :type language_slug: str
+
+        :return: true or false
+        :rtype: boolean
+        """
+        deepl_langs = [
+            "bg",
+            "cs",
+            "da",
+            "de",
+            "el",
+            "en",
+            "es",
+            "et",
+            "fi",
+            "fr",
+            "hu",
+            "it",
+            "ja",
+            "lt",
+            "lv",
+            "nl",
+            "pl",
+            "pt",
+            "ro",
+            "ru",
+            "sk",
+            "sl",
+            "sv",
+            "zh",
+        ]
+        if language_slug in deepl_langs:
+            return language_slug in deepl_langs
+        else:
+            messages.warning(
+                request,
+                _("This language isn't supported by DeepL. Please try another one"),
+            )
```

Comment:
```suggestion
        supported_source_languages = [
            source_language.code.lower()
            for source_language in self.translator.get_source_languages()
        ]
        supported_target_languages = [
            target_languages.code.lower()[:2]
            for target_languages in self.translator.get_target_languages()
        ]
        return (
            request.region.default_language.slug in supported_source_languages
            and language_slug in supported_target_languages
        )
```
I'd probably go with something along the lines of this - I think requesting the supported languages via the deepl api client is easy enough that we can directly implement it like this without the intermediary solution of defining constants.